libs/MVS/SceneTexture.cpp

```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
 		typedef TRasterMesh<RasterMesh> Base;
 		FaceMap& faceMap;
 		FIndex idxFace;
+		cv::Mat mask;
+		bool validFace;
+		std::vector<ImageRef> prevPixels;
```

Comment:
because we have to still handle depth-map rastering, and because there are very few faces invalid, compared to valid once, it is much more expensive to record pixels in this vector, compared to simply raster the face normally and only record if it touched invalid pixels in the mask, and if it did, call the face raster one more time, with a modified version that only sets the faceidx to -1